docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+  transaction. This is a required field
+* `DESCRIPTION` is a required field.
+* The `$AMOUNT` takes in a number, is also a required field.
+* `-c CATEGORY` is an optional field that takes in a user-defined category of the product
+* `-date DD/MM/YYYY` is an optional field that takes in the date of a transaction
+
+As this is quite a long command to parse, we will use regular expressions in the following steps to match and break
+down the instructions.
+
+1. function addStatement will call function returnRemainingInformation which will check whether the input contains the
+   optional -c and -d commands. This is done through the following:
+
+    - `-(in|out)\s+(.+)\$([\d.]+)` checks for the corresponding structure: `[-in/out] [whitepsace]
+      [description] [$amount] `. This will match when the optional flags are not included. An empty string will then be
+      returned.
+    - `-(in|out)\s+(.+)\$([\d.]+)\s+(.*)` checks for the corresponding structure `[-in/out] [whitepsace]
+      [description] [$amount] [remaining input]`. This will match when at least one of the optional flags are included,
+      and a string corresponding to `[remaining input]` will be returned.
+
+
+2. If an empty string is returned, an addCommand object will be returned from function addStatement. Otherwise,
+   a variable `String remainingInformation` will correspond to `[remaining input]` above. `remainingInformation` will
+   then
+   be checked for if it contains `-c` or `-date` flags.
+
+
+3. if `remainingInformation` contains a `-c` flag, it means that the user has provided a category for the item.
+   Thus, it will be passed into the function setCategory. setCategory will then use the following regex to match
+   `remainingInformation`:
+    - `-c\\s+(\\S+)` checks for the corresponding structure `[-c] [whitespace] [1-word category]`.
+    - `-c\\s+(\\S+)\\s+(.*)` checks for the same thing except category can be multiple words
+
+4. The last field to check is the `-date` field. If present, the setDate function will be called
+   on `remainingInformation`.
+   The setDate function will then use the following regex to match.
+    - `-date\\s+(\\d{2}/\\d{2}/\\d{4})` matches the corresponding
+      structure`[-date] [whitespace] [date in DD/MM/YYYY format]`
+
+These steps will ultimately parse the user's input and extract the necessary information. If any pattern does not match,
+our parser will throw a `RainDayException` indicating the wrong input format.
+
+#### Alternatives considered
+
+We have tried using the .split() function of strings to extract information. However, as we introduced more features and
+the commands grew more complex, the .split() function became very messy and inconvenient. Using this will also result in
+us needing to do more exceptions and error handling, which just made the entire process very complicated. Thus, we opted
+to use regular expressions, which is a more tidy and logical way to parse the inputs.
+
```

Comment:
The alternative method paragraph here is useful for developers who might consider going down this path!